src/js/localstorage.js

```diff
@@ -0,0 +1,187 @@
+//TESTING LOCALSTORAGE WHETHER IT WORKS OR EXIST 
+export function localStorageTest() {                 //to avoid name collisions
+    
+    try {
+        localStorage.setItem("test","test");
+        localStorage.removeItem("test");
+        console.log('Local Storage Fine')
+        return true;
+    } 
+    catch(e) {
+        return false;
+    }
+}
+
+//LOCKING THE START BUTTON IF LOCAL STORAGE DOESNT WOKR
+export const lockStart = () => {
+    if (localStorageTest()) {
+      return;
+    } 
+    else {
+      document.querySelector("a").remove();
+      document.querySelector("main").insertAdjacentHTML("beforeend",'<button class="btn-fail">Start</button>');
+      document.querySelector(".btn-fail").addEventListener("mouseover", function () {
+      document.querySelector("main").insertAdjacentHTML("beforeend",'<p class="err-p">Fix your local storage</p>');});
+      document.querySelector(".btn-fail").addEventListener("mouseout", function () {document.querySelector(".err-p").remove();});
+    }
+  };
+
+//SAVING TO AND LOADING FROM LOCALSTORAGE - universal function
+
+export function saveToLocalStorage(resource, positionName) {
+
+    const resourceToSave = JSON.stringify(resource);
+    localStorage.setItem(positionName, resourceToSave);
+
+} 
+
+export function loadFromLocalStorage(positionName) {
+
+    const resource = localStorage.getItem(positionName);
+    const resourceToLoad = JSON.parse(resource);
+    return resourceToLoad;
+}
+
+//MAIN OBJECTS , GETTERS AND SETTERS METHODS
+
+//============================================================
+export const setGame = function () {
+const game = {
+    questions: [],
+    currentQuestion: 0,
+    path: "",
+    tier: "",
+    }
+saveToLocalStorage(game,"game");
+}
+
+export const getGame = () => {
+    return loadFromLocalStorage("game");
+};
+
+//============================================================
+export const setPlayer = function () {
+const player = {
+    name:"",
+    skipUsed: false, // useSkip przestawia na true
+    score: 0,        // increase by 1 after correct answer
+    lives: 3,        // decrease by 1 after wrong answer
+    };
+saveToLocalStorage(player,"player");
+}
+
+export const getPlayer = () => {
+    return loadFromLocalStorage("player");
+};
+
+//============================================================
+
+export const setScoreboard = () => {
+    const scoreboard = [{name: '', score: 0}];
+
+saveToLocalStorage(scoreboard, "scoreboard");
+}
+
+const getScoreboard = () => {
+    return loadFromLocalStorage("scoreboard");
+};
+
+//API PARTICULAR FUNCTIONS
+
+const setQuestions = function (...ques){ //ques == questions, could be any type
```

Comment:
Do not name variable so nobody will be able to tell what is it by her name and then explain in the comment what it is.
You can just name the variable `questions`.
We do not need to tell what type it is. We should be able to tell that it is a list by reading a variable name.

Reply:
Done.